cli/command/stack/kubernetes/deploy.go

```diff
@@ -81,6 +74,29 @@ func RunDeploy(dockerCli *KubeCli, opts options.Deploy, cfg *composetypes.Config
 
 }
 
+func createResources(stack Stack, stacks StackClient, configMaps corev1.ConfigMapInterface, secrets corev1.SecretInterface) error {
+	var childResources []childResource
+
+	cr, err := stack.createFileBasedConfigMaps(configMaps)
+	childResources = append(childResources, cr...) // make sure we collect childresources already created in case of failure
+	if err != nil {
+		deleteChildResources(childResources)
```

Comment:
I'm not a fan of this deleter, I wonder if `createFileBasedConfigMaps` and `createFileBasedSecrets` could return the deleter itself instead?

Reply:
tried, but we still need childResources for setting owners, and that made the code uglier :(